sw/device/silicon_creator/lib/error.h

```diff
@@ -92,6 +92,7 @@ enum module_ {
   X(kErrorOtbnInvalidArgument,        ERROR_(1, kModuleOtbn, kInvalidArgument)), \
   X(kErrorOtbnBadOffsetLen,           ERROR_(2, kModuleOtbn, kInvalidArgument)), \
   X(kErrorOtbnExecutionFailed,        ERROR_(3, kModuleOtbn, kInternal)), \
+  X(kErrorOtbnUnavailable,            ERROR_(4, kModuleOtbn, kFailedPrecondition)), \
```

Comment:
Probably?

```suggestion
  X(kErrorOtbnUnavailable,            ERROR_(4, kModuleOtbn, kUnavailable)), \
```

The precondition checks in OTBN mostly (fully?) depend on the arguments themselves only, not on the state of the system.

Reply:
It's a bit awkward as the error codes and their definition ultimately comes from the google cloud RPC error codes.

The definition for `kUnavailable` is
> kUnavailable (gRPC code "UNAVAILABLE") indicates the service is currently unavailable and that this is most likely a transient condition.

Whilst the definition for `kFailedPrecondition` is

> kFailedPrecondition (gRPC code "FAILED_PRECONDITION") indicates that the operation was rejected because the system is not in a state required for the operation's execution.

I felt the latter better described what was going on.

Reply:
Ah ok, sounds good!